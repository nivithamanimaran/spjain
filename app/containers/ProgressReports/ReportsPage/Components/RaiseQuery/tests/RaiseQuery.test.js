import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import QueryPopupContent from '../index';
import ThemeProvider from '../../../../../../utils/themeProvider';
import { BrowserRouter as Router } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

describe('Query-Popup-Content', () => {
  const div = document.createElement('div');
  it('Query Popup Content renders correctly without props', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <QueryPopupContent />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <QueryPopupContent />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('query-popup-content')).toMatchSnapshot();
  });
  it('Query Popup Content renders correctly with props', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <QueryPopupContent
            rank={'10'}
            totalMarks={'660'}
            obtainedMarks={'440'}
          />
        </Router>
      </ThemeProvider>,
      div
    );

    const { queryByTestId } = render(
      <ThemeProvider>
        <Router>
          <QueryPopupContent
            rank={'10'}
            totalMarks={'660'}
            obtainedMarks={'440'}
          />
        </Router>
      </ThemeProvider>
    );

    const rank = queryByTestId('rank');
    expect(rank).toBeTruthy();
    expect(rank.textContent).toBe('10');

    const totalMarks = queryByTestId('total-marks');
    expect(totalMarks).toBeTruthy();
    expect(totalMarks.textContent).toBe(' headings:Division_Symbol660');

    const obtainedMarks = queryByTestId('obtained-marks');
    expect(obtainedMarks).toBeTruthy();
    expect(obtainedMarks.textContent).toBe('440 headings:Division_Symbol660');
  });
});
