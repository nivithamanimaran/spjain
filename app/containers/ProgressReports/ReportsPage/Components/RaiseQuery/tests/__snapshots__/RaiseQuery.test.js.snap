// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query-Popup-Content Query Popup Content renders correctly without props 1`] = `
<div
  class="MuiBox-root MuiBox-root-33"
  data-testid="query-popup-content"
>
  <div
    class="MuiBox-root MuiBox-root-34"
  >
    <div
      class="MuiGrid-root makeStyles-container-1 MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6"
      >
        <div
          class="MuiBox-root MuiBox-root-35 makeStyles-rank-5"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root makeStyles-firstItem-13 MuiGrid-item"
            >
              <p
                class="MuiTypography-root makeStyles-rankText-8 MuiTypography-body1"
              >
                headings:Rank
              </p>
              <span
                class="MuiTypography-root makeStyles-rankSubtext-9 MuiTypography-body1"
                data-testid="rank"
              />
              <span
                class="MuiTypography-root makeStyles-rankSubtext-9 makeStyles-rankSubSubtext-10 MuiTypography-body1"
              >
                headings:Rank_Suffix
              </span>
            </div>
            <div
              class="MuiGrid-root makeStyles-divider-20 MuiGrid-item"
            />
            <div
              class="MuiGrid-root makeStyles-item-14 MuiGrid-item"
            >
              <p
                class="MuiTypography-root makeStyles-rankText-8 MuiTypography-body1"
              >
                headings:Total_Marks
              </p>
              <span
                class="MuiTypography-root makeStyles-rankSubtext-9 MuiTypography-body1"
                data-testid="obtained-marks"
              >
                <span
                  class="MuiTypography-root makeStyles-rankSubtext-9 makeStyles-rankSubSubtext-10 makeStyles-markSubSubtext-11 MuiTypography-body1"
                  data-testid="total-marks"
                >
                   
                  headings:Division_Symbol
                  
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6"
      >
        <div
          class="MuiBox-root MuiBox-root-36 makeStyles-rank-5 makeStyles-promoted-6"
        >
          <p
            class="MuiTypography-root makeStyles-promotedText-12 MuiTypography-body1"
          >
            headings:Promoted_To_Text 
             
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-44"
  >
    <p
      class="MuiTypography-root makeStyles-textAreaLabel-16 makeStyles-sectionLabels-18 MuiTypography-body1"
    >
      headings:General_Remarks
    </p>
    <textarea
      aria-label="text-area"
      class="makeStyles-textArea-15"
      placeholder="labels:Query_Text_Area_Placeholder"
      rows="10"
      style="height: -4px; overflow: hidden;"
    />
    <textarea
      aria-hidden="true"
      class="makeStyles-textArea-15"
      readonly=""
      style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0);"
      tabindex="-1"
    />
  </div>
</div>
`;
